pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -1244,7 +1244,7 @@ func (ctrl *PersistentVolumeController) deleteVolumeOperation(volume *v1.Persist
 	// This method may have been waiting for a volume lock for some time.
 	// Previous deleteVolumeOperation might just have saved an updated version, so
 	// read current volume state now.
-	newVolume, err := ctrl.kubeClient.CoreV1().PersistentVolumes().Get(context.TODO(), volume.Name, metav1.GetOptions{})
+	newVolume, err := ctrl.volumeLister.Get(volume.Name)
```

Comment:
Could it race with a previous Delete operation that has just finished? In this case, the informer has an old PV object, while it was already deleted at the API server and in the storage backend.

In the worst case it seems PV controller will just call `Delete()` again, but our volume plugins were not tested thoroughly for idempotency and who knows what happens.
(With CSI migration getting GA for most of volume plugins that implement Delete(), do we even need to care?)